x-pack/plugins/observability_solution/observability/public/ui_actions/edit_slo_alerts_panel.tsx

```diff
@@ -49,18 +58,20 @@ export function createEditSloAlertsPanelAction(
         const result = await resolveEmbeddableSloUserInput(
           coreStart,
           pluginStart,
-          embeddable.getInput()
+          (embeddable as SLOAlertsEmbeddable).getSloAlertsConfig()
         );
-        embeddable.updateInput(result);
+        (embeddable as SLOAlertsEmbeddable).updateInput(result);
       } catch (e) {
         return Promise.reject();
       }
     },
-    async isCompatible({ embeddable }) {
-      return (
-        embeddable.type === SLO_ALERTS_EMBEDDABLE &&
-        embeddable.getInput().viewMode === ViewMode.EDIT
+    async isCompatible({ embeddable }: EmbeddableApiContext) {
```

Comment:
I was checking different PRs and it is not clear to me which style I should use. Should I rewrite to this instead?

```
async isCompatible(api: unknown | null): api is EditSloAlertsPanelApi =>
  Boolean( apiHasType(api) &&
          apiIsOfType(api, SLO_ALERTS_EMBEDDABLE) &&
          apiCanAccessViewMode(api) &&
          getInheritedViewMode(api) === ViewMode.EDIT)

Reply:
This distinction isn't very important. I prefer arrow functions `=>` because they keep their `this` [context](https://hostman.com/tutorials/arrow-functions-in-javascript/) no matter where they're called from, but in this instance it really doesn't matter and is up to personal preference.